### PR TITLE
[DOCS] HETERO pipeline parallelism

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
 - Document for HETERO pipeline parallelism""

### Tickets:
 - *CVS-138199*

### CODE REVIEW

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 1]
```diff
@@ -90,6 +100,30 @@ It does not take into account device peculiarities such as the inability to infe
          :language: cpp
          :fragment: [compile_model]
 
+Pipeline parallelism (preview)
```
Comment (peterchen-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:103]: @kblaszczak-intel  Is it correct to indicate this is a preview feature?

Comment (peterchen-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:103]:
```suggestion
Pipeline parallelism - preview
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 2]
```diff
@@ -66,7 +73,10 @@ Randomly selecting operations and setting affinities may lead to decrease in mod
 
 
 The Automatic Mode
```
Comment (tsavina) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:75]:
```suggestion
Automatic Mode
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 3]
```diff
@@ -66,7 +73,10 @@ Randomly selecting operations and setting affinities may lead to decrease in mod
...
-+++++++++++++++++++++
+++++++++++++++++++
+
+Without pipeline parallelism
```
Comment (tsavina) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:78]:
```suggestion
Without Pipeline Parallelism
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 4]
```diff
@@ -90,6 +100,30 @@ It does not take into account device peculiarities such as the inability to infe
          :language: cpp
          :fragment: [compile_model]
 
+Pipeline parallelism - preview
+-------------------------------
```
Comment (tsavina) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:104]:
```suggestion
Pipeline Parallelism (Preview)
--------------------------------
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 5]
```diff
@@ -90,6 +100,30 @@ It does not take into account device peculiarities such as the inability to infe
...
+
+The pipeline parallelism is set via ``ov::hint::model_distribution_policy``. This mode is an efficient technique to infer large models on multiple devices. The model is split into multiple stages, and each stage is assigned to a different device (``dGPU``, ``iGPU``, ``CPU``, etc.). This mode assign operations to different devices as reasonably as possible, ensuring that different stages can be executed in sequence and minimizing the amount of data transfer between different devices.
```
[the remark this answers is not in the document]
Comment (tsavina) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:106]: reasonably should be clarified here, does  it mean that the goal is to balance the load across devices? I suggest using evenly but feel free to change

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 6]
```diff
@@ -90,6 +100,30 @@ It does not take into account device peculiarities such as the inability to infe
...
+
+For large models which don’t fit on a single first priority device, model pipeline parallelism is employed where certain parts of the model are placed on different devices to ensure that the device has enough memory to infer these operations.
```
Comment (tsavina) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:108]:
```suggestion
For large models that do not fit on a single first-priority device, model pipeline parallelism is employed. This technique distributes certain parts of the model across different devices, ensuring that each device has enough memory to infer the operations.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 7]
```diff
@@ -18,6 +18,8 @@ Execution via the heterogeneous mode can be divided into two independent steps:
...
 2. Compiling a model to the Heterogeneous device assumes splitting the model to parts, compiling them on the specified devices (via `ov::device::priorities <https://docs.openvino.ai/2024/api/c_cpp_api/structov_1_1device_1_1_priorities.html>`__), and executing them in the Heterogeneous mode. The model is split to subgraphs in accordance with the affinities, where a set of connected operations with the same affinity is to be a dedicated subgraph. Each subgraph is compiled on a dedicated device and multiple `ov::CompiledModel <https://docs.openvino.ai/2024/api/c_cpp_api/classov_1_1_compiled_model.html#doxid-classov-1-1-compiled-model>`__ objects are made, which are connected via automatically allocated intermediate tensors.
+   
+   If you set pipeline parallel (via ``ov::hint::model_distribution_policy``), the model is split into multiple stages, and each stage is assigned to a different device. The output of one stage is fed as input to the next stage.
```
Comment (tsavina) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:22]:
```suggestion
   If you set pipeline parallelism (via ``ov::hint::model_distribution_policy``), the model is split into multiple stages, and each stage is assigned to a different device. The output of one stage is fed as input to the next stage.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst (hunk 1 above)
Comment (kblaszczak-intel) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst]: It surely is

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (peterchen-intel) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:106]: It is something like following, but let me check with @WeldonWangwang 
This mode assigns operations across devices proportionally per available memory size of each devices.

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (peterchen-intel) [reply docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst]: Duplicates the following one.

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst [hunk 8]
```diff
@@ -90,6 +100,30 @@ It does not take into account device peculiarities such as the inability to infe
          :language: cpp
          :fragment: [compile_model]
 
+Pipeline Parallelism (Preview)
+--------------------------------
+
+Pipeline parallelism is set via ``ov::hint::model_distribution_policy``. This mode is an efficient technique for inferring large models on multiple devices. The model is divided into multiple stages, with each stage assigned to a different device (``dGPU``, ``iGPU``, ``CPU``, etc.). This mode assigns operations across devices as evenly as possible, ensuring that different stages can be executed in sequence and minimizing data transfer between devices.
```
Comment (peterchen-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/hetero-execution.rst:106]:
```suggestion
Pipeline parallelism is set via ``ov::hint::model_distribution_policy``. This mode is an efficient technique for inferring large models on multiple devices. The model is divided into multiple stages, with each stage assigned to a different device (``dGPU``, ``iGPU``, ``CPU``, etc.). This mode assigns operations across devices proportionally per available memory size of each devices, ensuring that different stages can be executed in sequence and minimizing data transfer between devices.
```

**Review by peterchen-intel — APPROVED** [review]

**Review by tsavina — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**